rosetta-source/src/main/rosetta/event-common-func.rosetta

```diff
@@ -1661,6 +1661,87 @@ func Qualify_Shaping: <"The qualification of a shaping event from the fact that
 		and packageRef count = openTradeNoExecutionDetails count
 		and packageRef distinct count = 1
 
+
+func Create_PartialDeliveryPrimitiveInstruction: <"Creates the primitive instructions for a substitution of collateral by replacing the assetpayout of the trade.">
+	inputs:
+		//ICMA-P2 Instructions
+		tradeState TradeState (1..1) <"The original trade to be closed.">
+		effectiveDate AdjustableOrRelativeDate (1..1) <"The date to close and open a new position.">
+		terminationDate AdjustableOrRelativeDate (1..1) <"The new termination date.">
+		deliveredPriceQuantity PriceQuantity (1..*) <"The price and quantity of the remaining trade. ">
+		undeliveredPriceQuantity PriceQuantity (1..*) <"The price and quantity of the remaining trade. ">
+	output: 
+		instruction PrimitiveInstruction (1..1)
+
+	condition ContractualProduct: <"Only a contractual product can be rolled.">
+		tradeState -> trade -> tradableProduct -> product -> contractualProduct exists
+
+	set instruction -> split -> breakdown: <"Sets the first part of the split to be a termination instruction for the existing trade.">
+		[ Create_PrimitiveInstruction(
+			empty, empty, empty, empty,
+						Create_QuantityChangeInstruction(
+						deliveredPriceQuantity, 
+						QuantityChangeDirectionEnum -> Replace, 
+						tradeState->trade->tradableProduct->tradeLot->lotIdentifier
+			),
+			empty, empty, empty, empty,	empty, empty, empty)]
+
+	add instruction -> split -> breakdown: <"Sets the second part of the split to be a new contract with a new asset payout.">
+		[
+		Create_PrimitiveInstruction(
+			empty, empty, empty, empty,
+			Create_QuantityChangeInstruction(
+				undeliveredPriceQuantity,
+				QuantityChangeDirectionEnum -> Replace,
+				empty
+				),
+			empty, empty,
+			Create_PartialDeliveryInstructionTermsChange(
+				tradeState -> trade -> tradableProduct -> product -> contractualProduct,
+				effectiveDate, terminationDate
+				),
+			empty, empty, empty, empty
+			)
+		]
+
+func Create_PartialDeliveryInstructionTermsChange:
```

Comment:
What is this function for? How is it different from any other terms change? Also it has no comments to explain what it is doing